### ISSUE TITLE
Wrong path used when processing profile directory

### ISSUE THREAD

**Issue by twenster** [Beep6581/RawTherapee#6112]
*OS:* MacOS Big Sur
*RT:* 5.8 (february version)

RAWTherapee is install in /Applications (mandatory for it to run)

*Behaviour:*
Make any MacOS update (security). Once the update is applied, rebooted and is back on the desktop, RAWTherapee thorws the following error:
<img width="553" alt="RAWTherapee Reboot" src="https://user-images.githubusercontent.com/5445041/107851909-91856200-6e0d-11eb-9d15-2067864fbc35.png">

> FATAL ERROR!
> Creation of the user's processing profile directory "/Library/Application support/RAWTherapee/config/profiles" failed!

1. How come RAWTherapee is called after an update? (I've no insight of MacOS mechanisms)
2. /Library/Application support/ directory is not writable by the user. However ~/Library/Application support/ is

The script launched after a macOS update should look into the user's home, and not the System's one.
Note, that RAWTherapee is not start by the user (double click on the icon)...

**Comment by Benitoite** [comment]
Not sure about 5.8- it is a year old.

**Comment by twenster** [comment]
Which version is currently officially available to work with ? So I can have a look.

**Comment by Benitoite** [comment]
https://kd6kxr.keybase.pub/RawTherapee_OSX_10.12_64_5.8-2807-g31641f948.zip is my latest generation.

**Comment by Benitoite** [comment]
Most likely an error was thrown in the past and got stuck in a notification queue that expresses itself on boot.  I believe this line sent the error:

https://github.com/Benitoite/RawTherapee/blob/45c7b62db248b45a1b9b6d808a7271231efac34b/rtgui/options.cc#L113

Now to find where the path is being built.

**Comment by tonigi** [comment]
I got the same error. It manifests on login. What is really puzzling is what on earth starts `rawtherapee-bin` on user login. I can find no trace of it in any of the launch agents.

Edit: this is the only possible clue I found:

```
responsible pid = 979
responsible unique pid = 979
responsible path = /System/Library/CoreServices/Setup Assistant.app/Contents/Resources/mbfloagent
```

**Comment by Benitoite** [comment]
Most likely an error was thrown in the past and got stuck in a notification queue that expresses itself on boot.

**Comment by tonigi** [comment]
Yes, but where in MacOS is this queue? Is the notification queued automatically on "throw"? I couldn't find any documentation, nor file, on this regard. 

To clarify: the window is not a "system notification". It's rawtherapee-bin actually being executed.

**Comment by twenster** [comment]
From the look and feel of the window, MacOS doesn't render it. The font is Helvetica, and not the system font (San Francisco), also the OK button would not be that large, and not centered (but small on the right handside of the window), and there is no margin between the button and the window borders. The window look is much like the one rendered by RawTherapee istelf (ie GTK+ ?).

Also, not shown in the screenshot, the window was displayed at the top left of the screen. System message window are normally placed in the center of the screen.

I hoped I could find some logs in /var/log or /var/logs or even /usr/local/var/log. But nothing found.

EDIT: I've no experience in developping macos apps, I only explain with my experience in normal behaviour of the system.

**Comment by tonigi** [comment]
I think "something" is executing rawtherapee-bin as a system user. This makes it search for a missing path, and throw an exception which manifests with that gtk window. 

What that "something" is is a real mystery, and also a bit concerning if you ask me.

**Comment by twenster** [comment]
It looks like so for me too. I would expect a LaunchAgent or a LaunchDaemon (system or user), but couldn't find one. May be MacOS checks (registered) applications after a MacOS update, then RAWTherapee is and looks for system wide profiles instead of user profile. This window displayed after macos update... Only supposition.

**Comment by tonigi** [comment]
Apparently it's no Agent or Daemon. Seems something exotic. Hypotheses:

- color profiles
- an update of the photo library

**Comment by Benitoite** [comment]
Please confirm you all are testing the dev build cleanly and not a release nor on top of a previous install, and state your OS versions.  I cannot reproduce on 11β.

**Comment by tonigi** [comment]
No sorry, I am using release. It occurred upon several OS upgrades, and certainly in 11.2.2 -> 11.2.3

**Comment by Pandagrapher** [comment]
> I cannot reproduce on 11β.

Same for me: I don't have this behavior anymore when installing a Big Sur update with 5.8 release version.
Moreover, profile directory path computation has been updated in dev since 5.8. We should see if this issue is still available with future 5.9 release.

**Comment by vancleve** [comment]
I just ran into this.

I wonder if its a "saved application state" that can be found here
```
/Users/.../Library/Saved Application State/com.rawtherapee.rawtherapee.savedState
```
??
